app/src/main/java/com/example/spoonaapp/ui/searchview/SearchViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.spoonaapp.ui.searchview
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.spoonaapp.data.model.Product
+import com.example.spoonaapp.data.repository.ProductRepository
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.kotlin.addTo
+import io.reactivex.rxjava3.schedulers.Schedulers
+import java.util.*
+
+class SearchViewModel(
+    private val repository: ProductRepository
+) : ViewModel() {
+
+    private val disposables = CompositeDisposable()
+
+    private val _product = MutableLiveData<List<Product>>()
+    val product: LiveData<List<Product>> get() = _product
```

Comment:
Tại sao không dùng luôn MutableLiveData luông mà phải private làm gì ?

Reply:
E thấy thằng LiveData ko có setValue vì vậy cần thằng MutableLiveData để set giá trị. trong trường hợp không cần setData thì dùng luôn thằng liveData

Reply:
Tại sao lại không cần setData ?

Reply:
khai báo luôn như này thì sao ? 
```suggestion
    val product = MutableLiveData<List<Product>>()
```

Reply:
như vậy thì ta có thể chính sửa được dữ liệu ngay cả khi ở trong activity điều này ảnh hưởng đến bảo mất thông tin, vì ta chỉ muốn chỉnh sửa dữ liệu tại lớp model này thôi